src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -1902,9 +2010,88 @@ public async Task<bool> EnableExceptions(SessionId sessionId, string state, Canc
+                for (int j = 0; j < parmCount; j++)
+                {
+                    var monoTypeId = getCAttrsRetReader.ReadByte();
+                    if ((ValueTypeId)monoTypeId != ValueTypeId.Type)
+                        continue;
+                    var cAttrTypeId = getCAttrsRetReader.ReadInt32();
+                    var commandParams = new MemoryStream();
+                    var commandParamsWriter = new MonoBinaryWriter(commandParams);
+                    commandParamsWriter.Write(cAttrTypeId);
+                    var className = await GetTypeNameOriginal(sessionId, cAttrTypeId, token);
+                    if (className.IndexOf('[') > 0)
+                    {
+                        className = className.Remove(className.IndexOf('['));
+                        var assemblyId = await GetAssemblyIdFromType(sessionId, cAttrTypeId, token);
+                        var assemblyName = await GetAssemblyNameOriginal(sessionId, assemblyId, token);
+                        var typeToSearch = className;
+                        typeToSearch += "[["; //System.Collections.Generic.List`1[[System.Int32,mscorlib,Version=4.0.0.0,Culture=neutral,PublicKeyToken=b77a5c561934e089]],mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
+                        List<int> genericList = await GetGenericTypesFromType(sessionId, typeId, token);
+                        for (int k = 0; k < genericList.Count; k++)
+                        {
+                            var assemblyIdArg = await GetAssemblyIdFromType(sessionId, genericList[k], token);
+                            var assemblyNameArg = await GetAssemblyNameOriginal(sessionId, assemblyIdArg, token);
+                            var classNameArg = await GetTypeNameOriginal(sessionId, genericList[k], token);
+                            typeToSearch += classNameArg +", " + assemblyNameArg;
+                        }
+                        typeToSearch += "]]";
+                        typeToSearch +=  ", " + assemblyName;
+                        var genericTypeId = await GetTypeByName(sessionId, typeToSearch, token);
+                        methodId = await GetMethodIdByName(sessionId, genericTypeId, ".ctor", token);
+                    }
+                    else
+                        methodId = await GetMethodIdByName(sessionId, cAttrTypeId, ".ctor", token);
+                    invokeParamsWriter.Write((byte)ElementType.Object);
+                    invokeParamsWriter.Write(objectId);
+
+                    var retMethod = await InvokeMethod(sessionId, invokeParams.ToArray(), methodId, "methodRet", token);
+                    DotnetObjectId.TryParse(retMethod?["value"]?["objectId"]?.Value<string>(), out DotnetObjectId dotnetObjectId);
+                    var displayAttrs = await GetObjectValues(sessionId, int.Parse(dotnetObjectId.Value), GetObjectCommandType.WithProperties | GetObjectCommandType.FromDebuggerProxyAttribute, token);
```

Comment:
Instead of making `GetObjectValues` handle this special case, maybe call `GetDebuggerProxyAttribute` right here. And if it returns null, then call `GetObjectValues`?

And rename `GetObjectCommandType.FromDebuggerProxyAttribute` to  `GetObjectCommandType.ForDebuggerProxyAttribute`.
And remove `GetObjectCommandType.FromDebuggerDisplayAttribute`.

So, `GetObjectValues` can determine the args for `GetTypeFields`, and `CreateJArrayForProperties` based on that.